### PR TITLE
WIP: 159 nuclei template support

### LINKED ISSUE #159: As a user, I want to define nuclei custom jobs

### PR DESCRIPTION
**Pull request by lm-sec**
Closes #159 and #129 

This PR adds the support for Nuclei templates through Nuclei custom jobs. See documentation edits for details.

For the UI, it also adds the support for tabs in the code editor. That way, multiple files can be edited at the same time. The behaviour did not change for single file usage. The code and language parameters now always reflect the focused tab. In the case of multiple tabs, you can request the content of a specific tab, which is useful if you want to save all the content, for instance. The 'vs' and 'vs-dark' code editor themes support the tab colors.

Some refactoring in the backend as well, removing duplicated code and adding it as a helper function.

### CODE REVIEW

File: jobs/job-base-images/python/stalker_job_sdk/stalker_job_sdk/__init__.py [hunk 1]
```diff
@@ -79,25 +79,27 @@ def __init__(
...
+    _log("@warning", message)
 
 
 def log_error(message: str):
-    print(f"@error {message}")
+    _log("@error", message)
+    
+
+def _log(prefix: str, message: str):
+    lines = str(message).splitlines()
```
Comment (Aboisier) [inline jobs/job-base-images/python/stalker_job_sdk/stalker_job_sdk/__init__.py:102]: On ne voudrait pas plutôt escaper les newlines?

**Review by Aboisier — APPROVED** [review]
J'ai mis quelques suggestions mais LGTM!

File: jobs/job-base-images/python/nuclei/nuclei_finding_test.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        self.host_test_cases = test_cases.get("host")
+        self.port_test_cases = test_cases.get("port")
+            
+        super().__init__(a)
+
+    # def setUp(self):
+    #     print("setup, called before every test")
+
+    # def tearDown(self):
+    #     print("teardown, called after every test")
```
[the remark this answers is not in the document]
Comment (lm-sec) [reply jobs/job-base-images/python/nuclei/nuclei_finding_test.py:28]: Je l'ai laissé en guise de documentation, comme on a pas d'autres exemples de ce framework

File: jobs/job-base-images/python/stalker_job_sdk/stalker_job_sdk/__init__.py (hunk 1 above)
Comment (lm-sec) [reply jobs/job-base-images/python/stalker_job_sdk/stalker_job_sdk/__init__.py:102]: Ouais je voulais que ça fasse joli dans le UI, mais ça crée un million de findings, c'est prob mieux de juste escape et au pire modifier le UI si on est pas content

**Pull request merged**